### ISSUE TITLE
Estraverse 5.2.0 is not compatible with current ui5-builder

### ISSUE THREAD

**Issue by MrUnkei** [SAP/ui5-builder#490]
## Expected Behavior
The build work

## Current Behavior
Error message appear

## Steps to reproduce the issue

On a repo with `@ui5/cli@^2.0.0` and with `ui5 build xxxx` works
1. remove all your dependencies and delete package-lock.json file
2. `npm i` to use latest version of your dependencies
3. run your `ui5 build xxxx`

## Context
- UI5 Module Version *(output of `ui5 --version` when using the CLI)*: `{...}`
- Node.js Version: tested on `10.19.0` and `12.18.2`
- npm Version: `{...}`
- OS/Platform: `{...}`
- Browser *(if relevant)*: `{...}`
- Other information: `{...}`

## Affected components *(if known)*

- [X] [ui5-builder](https://github.com/SAP/ui5-builder)
- [ ] [ui5-server](https://github.com/SAP/ui5-server)
- [ ] [ui5-cli](https://github.com/SAP/ui5-cli)
- [ ] [ui5-fs](https://github.com/SAP/ui5-fs)
- [ ] [ui5-project](https://github.com/SAP/ui5-project)
- [ ] [ui5-logger](https://github.com/SAP/ui5-logger)

## Log Output / Stack Trace
```
Error Message:
taskRepository: Failed to require task module for generateComponentPreload: unknown estree node type 'ChainExpression', new syntax?

Stack Trace:
Error: taskRepository: Failed to require task module for generateComponentPreload: unknown estree node type 'ChainExpression', new syntax?
    at getTask (/XXXX/node_modules/@ui5/builder/lib/tasks/taskRepository.js:36:9)
    at addTask (/XXXX/node_modules/@ui5/builder/lib/types/application/ApplicationBuilder.js:85:12)
    at /XXXX/node_modules/@ui5/builder/lib/types/AbstractBuilder.js:205:11
    at process._tickCallback (internal/process/next_tick.js:68:7)

```

### More information
Estraverse@5.2.0 has been release yesterday and introduce `ChainExpression`
https://github.com/estools/estraverse/pull/113

**Comment by RandomByte** [comment]
Related: https://github.com/SAP/ui5-builder/issues/309#issuecomment-521108883

**Comment by danielang** [comment]
When a new version of the `@ui5/cli` will be realesed where the version of estraverse will be fixed?

Currently this will braek our buildserver and fresh installs of our dev environment/project. We use the exact version `2.3.1` of the `@ui5/cli`.

Thank you.

**Comment by MrUnkei** [comment]
Hi @danielang
on our side, we have a workaround, we had Estraverse@5.2.0 in our devDependencies, until to have a new version of @ui5/builder

**Comment by RandomByte** [comment]
> When a new version of the `@ui5/cli` will be realesed where the version of estraverse will be fixed?
> 
> Currently this will braek our buildserver and fresh installs of our dev environment/project. We use the exact version `2.3.1` of the `@ui5/cli`.

@danielang This should not happen. `@ui5/cli` comes with an [`npm-shrinkwrap.json`](https://github.com/SAP/ui5-cli/blob/v2.3.1/npm-shrinkwrap.json#L2882-L2886) which only allows `estraverse@5.1.0` to be installed for the tooling.

I also could not reproduce any issues with the CLI in the [openui5-sample-app](https://github.com/SAP/openui5-sample-app) after removing `node_modules` and `package-lock.json` and reinstalling all dependencies (including `@ui5/cli`).

How do you install `@ui5/cli`?

**Comment by danielang** [comment]
@RandomByte and @MrUnkei 
thank you for your replies.
We are  using a yarn workspace. So we install our dependencies with yarn.

A simple research shows that yarn does not support the npm-shrinkwrap file: [migrating-from-npm](https://classic.yarnpkg.com/en/docs/migrating-from-npm/):

> If you are using an npm-shrinkwrap.json file right now, be aware that you may end up with a different set of dependencies. Yarn does not support npm shrinkwrap files as they don’t have enough information in them to power Yarn’s more deterministic algorithm. If you are using a shrinkwrap file it may be easier to convert everyone working on the project to use Yarn at the same time. Simply remove your existing npm-shrinkwrap.json file and check in the newly created yarn.lock file.

What can we do?

**Comment by RandomByte** [comment]
I see. In that case, the `yarn.lock` file should prevent you from running into this issue. In case you have already updated this lockfile or you don't use it, you can also use [`resolutions`](https://classic.yarnpkg.com/en/docs/selective-version-resolutions/) to achieve something similar to what @MrUnkei explained.

**Comment by RandomByte** [comment]
Anyways, I guess we'll also do a `@ui5/cli` release for those Yarn users once we have a new `@ui5/builder` version. Thank you for the feedback.

**Comment by danielang** [comment]
Thank you too!

**Comment by RandomByte** [comment]
Pending release

**Comment by RandomByte** [comment]
Resolved with `@ui5/builder` [v2.0.7](https://github.com/SAP/ui5-builder/releases/tag/v2.0.7), which is contained in `@ui5/cli` [v2.3.2](https://github.com/SAP/ui5-cli/releases/tag/v2.3.2).
